### PR TITLE
Add genericpath stub

### PR DESCRIPTION
**Pull request by ethanhs**
To backport `os.path.commonpath` in mypy I needed to use genericpath. It seems unchanged since 3.4, and the `same*` functions were added in 3.4. (checked via comparing the __all__s of the source in 2.7, 3.4, and 3.7.

### CODE REVIEW

File: stdlib/2and3/genericpath.pyi [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+def exists(path: str) -> bool: ...
+def isfile(path: str) -> bool: ...
+def isdir(s: str) -> bool: ...
+def getsize(filename: str) -> int: ...
+def getmtime(filename: str) -> float: ...
+def getatime(filename: str) -> float: ...
+def getctime(filename: str) -> float: ...
+def commonprefix(m: List[str]) -> str: ...
+
+if sys.version_info[0] >= 3.4:
```
Comment (gvanrossum) [inline stdlib/2and3/genericpath.pyi]: That's not how you check versions.

Comment (ethanhs) [reply stdlib/2and3/genericpath.pyi]: Yeah... I don't really know what I was thinking there. Apologies.

**Comment by ilevkivskyi** [comment]
> after review that the differences between it and the new one are uninteresting

I have never used this module, so I am not sure which signatures should be used (almost all of them are different, but it looks like most of them can be fixed with `Text`). @ethanhs what do you think?

**Comment by gvanrossum** [comment]
Maybe just commonpath needs to have one definition for Py2 and another for Py3?

**Comment by ethanhs** [comment]
Yes, I think switching to `Text` will be the best. I'll make a PR now.

**Pull request merged**